clients/src/main/java/org/apache/kafka/common/security/ssl/OpenSslContextProvider.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.security.ssl;
+
+import java.security.NoSuchAlgorithmException;
+import java.security.Provider;
+import java.util.Map;
+import javax.net.ssl.SSLContext;
+import org.apache.kafka.common.config.SslConfigs;
+import org.conscrypt.Conscrypt;
+
+/**
+ * A OpenSslContext provider based on Conscrypt library.
+ */
+public class OpenSslContextProvider implements SslContextProvider {
+    private String protocol;
+    private Provider provider;
+
+    @Override
+    public void configure(Map<String, ?> configs) {
+        protocol = (String) configs.get(SslConfigs.SSL_PROTOCOL_CONFIG);
```

Comment:
the protocol here cannot be null in current code base, since the [configs passed in here is `AbstractConfig.values()`](https://github.com/linkedin/kafka/blob/7454dc9b198e3d7be2abd3344c6ba4d1eb1b4026/clients/src/main/java/org/apache/kafka/common/network/ChannelBuilders.java#L98), which is the parsed configs based on the config definition and [we have defined default value in `SslConfigs`](https://github.com/linkedin/kafka/blob/7454dc9b198e3d7be2abd3344c6ba4d1eb1b4026/clients/src/main/java/org/apache/kafka/common/config/SslConfigs.java#L125). 

I feel even this does not hold in the future, this will not cause big issue, we will get `NoSuchAlgorithmException` when calling `getSSLContext()`.